src/Stratis.Bitcoin/Base/ChainState.cs

```diff
@@ -66,39 +50,5 @@ public void MarkBlockInvalid(uint256 hashBlock, DateTime? rejectedUntil = null)
         {
             this.invalidBlockHashStore.MarkInvalid(hashBlock, rejectedUntil);
         }
-
-        /// <summary>
-        /// This method will check if the node is in a state of IBD (Initial Block Download)
-        /// </summary>
-        public bool IsInitialBlockDownload
```

Comment:
So where did this logic move to? If Consensus is not present then IBD is meaningless.

Reply:
IBD logic is now accessible only through the IBlockDownloadState. I've replaced all usages of that property with accessing instance of IBlockDownloadState.

Reply:
OK but there is some logic here like checking the ibdLastUpdate as well as checking that the ibdStateProvider (in this instance, Consensus) exists. If not we return false. Where is this logic in the new BlockDownloadStateProvider? Sorry if i'm missing something here?